### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by DolceTriade**

### CODE REVIEW

File: backend/notifications.py [hunk 1]
```diff
@@ -26,12 +28,19 @@ def ExecuteNotifications(request, firebase):
...
+  the query first.
   """
   # Handle notifications, 20 at a time so we don't need to get the entire
   # notification database.
   get_params = {
-      'orderBy': '"sent"',
-      'limitToLast': 20
+      'orderBy': '"sendTime"',
+      'limitToLast': 20,
+      'endAt': int(time.time())
```
Comment (IsaacG) [inline backend/notifications.py]: Note you call in(time.time()) twice in two lines

File: backend/notifications.py [hunk 2]
```diff
@@ -40,10 +49,10 @@ def ExecuteNotifications(request, firebase):
     if notifications is None:
       return
     for notification_id, notification in notifications.iteritems():
-      if 'sent' not in notification and current_time > int(notification['sendTime']):
+      if notification['sendTime'] > 0 and current_time > int(notification['sendTime']):
```
Comment (IsaacG) [inline backend/notifications.py:52]: You can test notification ['sendTime'] without the >0 bit.

File: backend/backend_test_data.json [hunk 3]
```diff
@@ -130,7 +130,18 @@
...
-  "notifications": {},
+  "notifications": {
+    "notification-$(ident)s-abc": {
+        "groupId": "group-%(ident)s",
+        "message": "hello",
+        "app": true,
+        "vibrate": false,
+        "sound": "none",
+        "destination": "https://google.com",
+        "sendTime": 1494381500,
```
Comment (IsaacG) [inline backend/backend_test_data.json]: This is going to foul up tests quickly. See the clean data method in the test file to handle this.

**Review by IsaacG — COMMENTED** [review]
Looks good!

File: backend/backend_test_data.json (hunk 3 above)
Comment (DolceTriade) [reply backend/backend_test_data.json]: Just removed until i write tests for real.

File: backend/notifications.py (hunk 2 above)
Comment (DolceTriade) [reply backend/notifications.py:52]: I was just being pedantic. Someone could alter the database directly with a negative value or something.

File: backend/notifications.py (hunk 1 above)
Comment (DolceTriade) [reply backend/notifications.py]: Good catch. fixed.

**Review by IsaacG — APPROVED** [review]
LGTM

**Pull request merged**